### PR TITLE
Review and refactor the actions, reducers and middleware for transaction - Closes #2176

### LINKED ISSUE #2176: Review and refactor the actions, reducers and middleware for transaction
### Descriptions
This is a child ticket for #2020.
With a short review of the code, we can find several redundant or obsolete actions. accordingly, there are cases defined in the reducers switch-case statement which can be refactored or even removed. 
The naming of the actions seems to need revision.

### PR DESCRIPTION
**Pull request by osvaldovega**
### What issue have I solved?

-- #2176 

### How have I implemented/fixed it?

- [x] Improve naming for transactions actions functions.
- [x] Replace promises to use async/await
- [x] Add comments to understand better the functions
- [x] Add TODO comments in the reducer and actions file for remove once get single transaction data, create tx and broadcast tx code be part of utils.
- [x] Update all unit tests file and components to use the new names

### How has this been tested?

1. Login into the app.
2. Check dashboard data load properly (recent tx).
3. Go to wallet see all data load properly (all tx, incoming and outgoing) without errors
4. Check any tx and see the details of the tx

In summary all the app should continue working as before without any problem

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/transactions/explorerTransactions/explorerTransactions.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ class ExplorerTransactions extends React.Component {
       activeToken: this.props.activeToken,
       transactions: this.props.transactions.data.data,
       activeFilter: this.props.transactions.urlSearchParams.filters.direction,
-      loading: this.props.transactions.isLoading ? [actionTypes.transactionsLoaded] : [],
+      loading: this.props.transactions.isLoading ? [actionTypes.getTransactionsSuccess] : [],
```
Comment (slaweet) [inline src/components/transactions/explorerTransactions/explorerTransactions.js]: Should be the same action as in https://github.com/LiskHQ/lisk-hub/blob/da0946a4f0b54281e404d732cc2fb501c6bf5904/src/actions/transactions.js#L65
```suggestion
      loading: this.props.transactions.isLoading ? [actionTypes.getTransactions] : [],
```

File: src/components/transactions/transactionsList.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ class TransactionsList extends React.Component {
     const filteredTransactions = transactions.filter(fixIncomingFilter);
 
     const isLoading = loading.filter(type =>
-      actionTypes.transactionsLoaded === type).length > 0;
+      actionTypes.getTransactionsSuccess === type).length > 0;
```
Comment (slaweet) [inline src/components/transactions/transactionsList.js]: Should be the same action as in https://github.com/LiskHQ/lisk-hub/blob/da0946a4f0b54281e404d732cc2fb501c6bf5904/src/actions/transactions.js#L65
```suggestion
      actionTypes.getTransactions === type).length > 0;
```

because otherwise the loading bar inside the transactions box on wallet page is not displayed when loading.

File: src/components/transactions/walletTransactions/walletTransactions.test.js [hunk 3]
```diff
@@ -68,8 +68,9 @@ describe('WalletTransactions Component', () => {
     transactionsCount: 1000,
     transaction: transactions[0],
     transactions,
-    loadTransactions: jest.fn(),
-    loadLastTransaction: jest.fn(),
+    getTransactions: jest.fn(),
+    getLastTransaction: jest.fn(),
+    updateAccountDelegateStats: jest.fn(),
```
Comment (slaweet) [inline src/components/transactions/walletTransactions/walletTransactions.test.js]: `updateAccountDelegateStats` should be removed here - it was removed in https://github.com/LiskHQ/lisk-hub/pull/2299

File: src/store/reducers/transactions.js [hunk 4]
```diff
@@ -56,38 +63,42 @@ const transactions = (state = initialState, action) => { // eslint-disable-line
     case actionTypes.updateTransactions:
       return {
         ...state, // Filter any newly confirmed transaction from pending
-        pending: state.pending.filter(pendingTransaction =>
-          action.data.confirmed.filter(transaction =>
-            transaction.id === pendingTransaction.id).length === 0),
+        pending: merge(state.pending, action.data.confirmed),
```
Comment (slaweet) [inline src/store/reducers/transactions.js]: `merge` is not what this filter was doing. It returns all transactions from `array1` that are not in `array2`, based on `id`. I cannot think of a good name for it though ☹️

File: src/actions/transactions.js [hunk 5]
```diff
@@ -79,97 +100,113 @@ export const loadTransactions = ({
  *
  * @param {String} address - address of the active account
  */
-export const loadLastTransaction = address => (dispatch, getState) => {
+export const getLastTransaction = address => async (dispatch, getState) => {
```
Comment (slaweet) [inline src/actions/transactions.js]: The comment above says "This action is used to get the data for "My Wallet Details" module on wallet page". We don't have the last transaction in "My Wallet Details" module on wallet page anymore. IMO this action can be removed.

Comment (osvaldovega) [reply src/actions/transactions.js]: thx @slaweet for the review I was checking the code and the problem is that this function is still in use in.

https://github.com/LiskHQ/lisk-hub/blob/da0946a4f0b54281e404d732cc2fb501c6bf5904/src/components/transactions/walletTransactions/walletTransactions.js#L55

And because of this I can't remove.

I checked in development and is still in use

Comment (slaweet) [reply src/actions/transactions.js]: `getLastTransaction` is called from `walletTransactions`, but AFAIK that call can be removed as well. If you search for `getLastTransaction` in the whole project, you'll see that there is no reducer listening to the `getLastTransaction` action.

```
$ git grep getLastTransaction
src/actions/transactions.js:export const getLastTransaction = address => async (dispatch, getState) => {
src/actions/transactions.js:  dispatch(loadingStarted(actionTypes.getLastTransaction));
src/actions/transactions.js:  dispatch(loadingFinished(actionTypes.getLastTransaction));
src/components/transactions/explorerTransactions/explorerTransactions.test.js:    getLastTransaction: jest.fn(),
src/components/transactions/walletTransactions/index.js:import { getLastTransaction, getTransactions } from '../../../actions/transactions';
src/components/transactions/walletTransactions/index.js:  getLastTransaction,
src/components/transactions/walletTransactions/walletTransactions.js:    this.props.getLastTransaction(this.props.account.address);
src/components/transactions/walletTransactions/walletTransactions.test.js:    getLastTransaction: jest.fn(),
src/constants/actions.js:  getLastTransaction: 'GET_LAST_TRANSACTION',
```

Comment (slaweet) [reply src/actions/transactions.js]: I pushed a commit to remove it https://github.com/LiskHQ/lisk-hub/pull/2312/commits/04b8f820773f2494126280fdab2fadacc612a328

**Comment by slaweet** [comment]
Based on the standup discussion, merging this one and creating a follow-up https://github.com/LiskHQ/lisk-hub/issues/2316
